### PR TITLE
Adds support for publishing docs to pulpproject.org

### PR DESCRIPTION
**Pull request by dkliban**
re: #7229
https://pulp.plan.io/issues/7229

### CODE REVIEW

File: templates/travis/.travis/docs-builder.py.j2 [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    ga_build = False
+
+    if not re.search("[a-zA-Z]", branch) and len(branch.split(".")) > 2:
+        ga_build = True
+
+    # build the docs via the Pulp project itself
+    print("Building the docs")
+    docs_directory = os.sep.join([WORKING_DIR, "docs"])
+
+    make_command = ["make", "PULP_URL=http://pulp", "diagrams", "html"]
```
Comment (fao89) [inline templates/travis/.travis/docs-builder.py.j2:89]: Not about this PR, but I think we should have `make bindings_docs`

Comment (dkliban) [reply templates/travis/.travis/docs-builder.py.j2:89]: yes, we will add that separately.

**Comment by daviddavis** [comment]
Code LGTM. What about some docs in the README and also a changelog entry?

**Review by daviddavis — COMMENTED** [review]
Thanks for adding a changelog and README info. Looks great.

**Review by daviddavis — APPROVED** [review]

**Pull request merged**